services/notification/__init__.py

```diff
@@ -101,39 +116,52 @@ def get_notifiers_instances(self) -> Iterator[AbstractBaseNotifier]:
 
         current_flags = [rf.flag_name for rf in self.repository.flags if not rf.deleted]
         for key, title, status_config in self.get_statuses(current_flags):
-            status_notifier_class = get_status_notifier_class(key, "status")
-            if self._should_use_checks_notifier():
-                checks_notifier = get_status_notifier_class(key, "checks")
-                yield ChecksWithFallback(
-                    checks_notifier=checks_notifier(
-                        repository=self.repository,
-                        title=title,
-                        notifier_yaml_settings=status_config,
-                        notifier_site_settings={},
-                        current_yaml=self.current_yaml,
-                        decoration_type=self.decoration_type,
-                        gh_installation_name_to_use=self.gh_installation_name_to_use,
-                    ),
-                    status_notifier=status_notifier_class(
+            should_use_status_notifiers = self._should_use_status_notifier(
+                status_type=key
+            )
+            should_use_checks_notifier = self._should_use_checks_notifier(
+                status_type=key
+            )
+
+            # Introduced to gate team plan not having statuses nor checks based on conditionals.
+            if not should_use_status_notifiers and not should_use_checks_notifier:
+                continue
+
+            # We always send statuses, so there's currently no case of checks without status.
+            if should_use_status_notifiers:
```

Comment:
for readability maybe the logic following here should be encapsulated in a function